### PR TITLE
Update to Gradle 8.5

### PR DESCRIPTION
**Pull request by reta**
### Description
Update to Gradle 8.5

### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/10334

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,42 +59,65 @@ subprojects {
...
-        runtimeClasspath("com.oracle.oci.sdk:oci-java-sdk-objectstorage:${sdk_version}")
-        testRuntimeClasspath("com.oracle.oci.sdk:oci-java-sdk-objectstorage:${sdk_version}")
-
-        runtimeClasspath("org.slf4j:slf4j-api") {
-            version {
-                strictly "1.7.33"
-            }
-        }
-        testRuntimeClasspath("org.slf4j:slf4j-api") {
+        implementation("com.oracle.oci.sdk:oci-java-sdk-common:${sdk_version}")
```
Comment (reta) [inline build.gradle:88]: Those are not new dependencies, all of them were brought in transitively

Comment (samuel-oci) [reply build.gradle:88]: Got it, thanks @reta approved.

**Review by samuel-oci — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (reta) [reply build.gradle:88]: @samuel-oci do you mind approving the workflow? seems like few checks need to be run, thank you

Comment (samuel-oci) [reply build.gradle:88]: done

Comment (reta) [reply build.gradle:88]: Apologies, @samuel-oci may I ask you to approve one more time (fixed build issue), thank you

**Pull request merged**